tests/integration/ceph_osd_on_pvc_test.go

```diff
@@ -0,0 +1,160 @@
+	defer func(s *OSDOnPVCSuite) {
+		HandlePanics(recover(), s.op, s.T)
+	}(s)
+	suite.Run(t, s)
+}
+
+type OSDOnPVCSuite struct {
+	suite.Suite
+	helper    *clients.TestClient
+	op        *TestCluster
+	k8sh      *utils.K8sHelper
+	namespace string
+}
+
+// createPVC creates a PVC for a OSD.
+func (suite *OSDOnPVCSuite) createPVC() {
+	cmdArgs := utils.CommandArgs{Command: localPathPVForMonCmd, CmdArgs: []string{}}
+	cmdOut := utils.ExecuteCommand(cmdArgs)
+	require.NoError(suite.T(), cmdOut.Err)
+	cmdArgs = utils.CommandArgs{Command: localPathPVForOSDCmd,
+		CmdArgs: []string{installer.TestScratchDevice(), sys.DiskType, "create"}}
+	cmdOut = utils.ExecuteCommand(cmdArgs)
+	require.NoError(suite.T(), cmdOut.Err)
+}
+
+// Deploy a Rook cluster
+func (suite *OSDOnPVCSuite) SetupSuite() {
+	suite.namespace = "osd-on-pvc-ns"
+	osdOnPVCTestCluster := TestCluster{
+		namespace:               suite.namespace,
+		storeType:               "bluestore",
+		storageClassName:        "manual",
+		useHelm:                 false,
+		usePVC:                  true,
+		mons:                    3,
+		rbdMirrorWorkers:        1,
+		rookCephCleanup:         true,
+		skipOSDCreation:         false,
+		minimalMatrixK8sVersion: osdOnPVCSuiteMinimalTestVersion,
+		rookVersion:             installer.VersionMaster,
+		cephVersion:             installer.OctopusVersion,
+	}
+	suite.createPVC()
+	suite.op, suite.k8sh = StartTestCluster(suite.T, &osdOnPVCTestCluster)
+	suite.helper = clients.CreateTestClient(suite.k8sh, suite.op.installer.Manifests)
+}
+
+func (suite *OSDOnPVCSuite) AfterTest(suiteName, testName string) {
+	suite.op.installer.CollectOperatorLog(suiteName, testName, installer.SystemNamespace(suite.namespace))
+}
+
+func (suite *OSDOnPVCSuite) TearDownSuite() {
+	suite.op.Teardown()
+}
+
+func (suite *OSDOnPVCSuite) createOSD(deviceType string) {
+	cmdArgs := utils.CommandArgs{Command: localPathPVForOSDCmd,
+		CmdArgs: []string{installer.TestScratchDevice(), deviceType, "create"}}
+	cmdOut := utils.ExecuteCommand(cmdArgs)
+	require.NoError(suite.T(), cmdOut.Err)
+
+	_, err := suite.k8sh.Kubectl("-n", suite.namespace, "patch",
+		"CephCluster", suite.op.installer.ClusterName,
+		"--type=json", "-p", `[{"op": "replace", "path": "/spec/storage/storageClassDeviceSets/0/count", "value":1}]`)
+	require.NoError(suite.T(), err)
+	err = suite.k8sh.WaitForPodCount("app=rook-ceph-osd", suite.namespace, 1)
+	require.Nil(suite.T(), err)
+}
+
+func (suite *OSDOnPVCSuite) deleteOSD(deviceType string) {
+	_, err := suite.k8sh.Kubectl("-n", suite.namespace, "patch", "CephCluster", suite.op.installer.ClusterName,
+		"--type=json", "-p", `[{"op": "replace", "path": "/spec/storage/storageClassDeviceSets/0/count", "value":0}]`)
+	require.NoError(suite.T(), err)
+	err = suite.k8sh.WaitForPodCount("app=rook-ceph-osd", suite.namespace, 0)
+	require.Nil(suite.T(), err)
+	_, err = suite.k8sh.Exec(suite.namespace, "rook-ceph-tools", "ceph", []string{"osd", "down", "osd.0"})
```

Comment:
@xenolinux After this test is merged, with #5943 you could update this test to call the new job instead of all these individual test commands. This would give you fairly simple integration test coverage on that change.

Reply:
Noted!